ReactWindows/ReactNative.Shared/ReactInstanceManager.cs

```diff
@@ -257,7 +272,7 @@ public void OnSuspend()
         public void OnEnteredBackground()
         {
             DispatcherHelpers.AssertOnDispatcher();
-            MoveToBackgroundLifecycleState();
+            MoveToBackgroundLifecycleState(false);
```

Comment:
Do you think it would help if we moved the whole state machine to a new class file? ReactInstanceManager is a monolith as is.